src/Type/TypeSerializer.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Efabrica\PHPStanLatte\Type;
+
+use PHPStan\PhpDoc\TypeStringResolver;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\Type;
+use PHPStan\Type\VerbosityLevel;
+use Throwable;
+
+class TypeSerializer
+{
+    private TypeStringResolver $typeStringResolver;
+
+    /**
+     * @param TypeStringResolver $typeStringResolver
+     */
+    public function __construct(TypeStringResolver $typeStringResolver)
+    {
+        $this->typeStringResolver = $typeStringResolver;
+    }
+
+    /**
+     * @param Type $type
+     * @return array<string, string>
```

Comment:
shouldn't be `array{precise: string, basic: string, serialized?: string}`? And also everywhere where it is used in value object array types

Reply:
My idea was taht exact format of serialized type is internal information of TypeSerializer - we might change/add keys and it never should be accessed outside this class

Reply:
Ok if we don’t need it outside of this class we can leave it string, string